[14.0][FIX] shopfloor_single_product_transfer: check package

mt-software-de opened the pull request:
check scanned package

mt-software-de commented:
cc @jbaudoux @mmequignon

jbaudoux reviewed (approved):
LGTM

jbaudoux commented:
@mmequignon Can you drop a review?

Ref: rau-235

simahawk commented on shopfloor_single_product_transfer/services/single_product_transfer.py:
```diff
@@ -718,7 +733,7 @@ def _set_quantity__by_package(self, move_line, package, confirmation=False):
             handlers = self._set_quantity__by_location_handlers()
             response = self._use_handlers(
-                handlers, move_line, location, confirmation=confirmation
+                handlers, move_line, location, package, confirmation=confirmation
```
```suggestion
                handlers, move_line, location, package=package, confirmation=confirmation
```

simahawk commented on shopfloor_single_product_transfer/services/single_product_transfer.py:
```diff
@@ -882,10 +897,10 @@ def set_location(self, selected_line_id, package_id, barcode):
         if handler:
-            return handler(move_line, search_result.record)
-        package = self.env["stock.quant.package"].browse(package_id)
+            return handler(move_line, search_result.record, package)
```
```suggestion
            return handler(move_line, search_result.record, package=package)
```

simahawk commented on shopfloor_single_product_transfer_force_package/services/single_product_transfer.py:
```diff
@@ -8,15 +9,20 @@
             )
-        return super()._set_quantity__check_location(move_line, location, confirmation)
+        return super()._set_quantity__check_location(
+            move_line, location, package, confirmation
```
```suggestion
            move_line, location, package=package, confirmation=confirmation
```

simahawk commented on shopfloor_single_product_transfer_unique_order_at_location/services/single_product_transfer.py:
```diff
@@ -10,9 +11,11 @@ class ShopfloorSingleProductTransfer(Component):
     ):
-        res = super()._set_quantity__check_location(move_line, location, confirmation)
+        res = super()._set_quantity__check_location(
+            move_line, location, package, confirmation
```
```suggestion
            move_line, location, package=package, confirmation=confirmation
```

simahawk reviewed (approved):
LG

mmequignon reviewed (commented):
LG, however instead of changing the signature for both `set_quantity` and `set_destination`, I would've created a dedicated handler.
Would be easier to modify the behavior in one enpoint only.

mt-software-de committed:
[FIX] shopfloor_single_product_transfer: Package scan

    Do not remove result_package_id in call of _set_quantity__by_location
    when it was previously set by scanning a package

mt-software-de commented:
@simahawk could you merge this one?

mt-software-de commented:
> @simahawk could you merge this one?

or maybe @jbaudoux ?

jbaudoux commented:
/ocabot merge minor

mt-software-de commented:
ping @jbaudoux  could you retry to merge this?

jbaudoux commented:
/ocabot merge minor
